policyengine_api/services/household_service.py

```diff
@@ -0,0 +1,128 @@
+import json
+from sqlalchemy.engine.row import LegacyRow
+
+from policyengine_api.data import database
+from policyengine_api.utils import hash_object
+from policyengine_api.constants import COUNTRY_PACKAGE_VERSIONS
+
+
+class HouseholdService:
+
+    def get_household(self, country_id: str, household_id: int) -> dict | None:
+        """
+        Get a household's input data with a given ID.
+
+        Args:
+            country_id (str): The country ID.
+            household_id (int): The household ID.
+        """
+        print("Getting household data")
+
+        try:
+            row: LegacyRow | None = database.query(
+                f"SELECT * FROM household WHERE id = ? AND country_id = ?",
+                (household_id, country_id),
+            ).fetchone()
+
+            print(row)
+            print(type(row))
+
+            # If row is present, we must JSON.loads the household_json
```

Comment:
**followon/nonblocking** We could do the same kind of validation/injest here for the database JSON as we input validation above. Bonus: return a real data type instead of a dict.

Reply:
That certainly makes sense. Just to note: one thing I'd love to do (I have an issue open somewhere) is also migrate to using a db ORM, probably SQLAlchemy v.2, so curious how this would interplay with that.